numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -961,48 +961,22 @@ string_intp_output_resolve_descriptors(
     return NPY_NO_CASTING;
 }
 
-template <IMPLEMENTED_UNARY_FUNCTIONS f, typename T>
-struct call_buffer_function {
-    void operator()(const char *buffer, size_t size, char *out) {
-        Buffer<ENCODING::UTF8> buf((char *)buffer, size);
-        switch (f) {
-            case IMPLEMENTED_UNARY_FUNCTIONS::ISALPHA:
-                *(T *)out = buf.isalpha();
-                break;
-            case IMPLEMENTED_UNARY_FUNCTIONS::ISDECIMAL:
-                *(T *)out = buf.isdecimal();
-                break;
-            case IMPLEMENTED_UNARY_FUNCTIONS::ISDIGIT:
-                *(T *)out = buf.isdigit();
-                break;
-            case IMPLEMENTED_UNARY_FUNCTIONS::ISNUMERIC:
-                *(T *)out = buf.isnumeric();
-                break;
-            case IMPLEMENTED_UNARY_FUNCTIONS::ISSPACE:
-                *(T *)out = buf.isspace();
-                break;
-            case IMPLEMENTED_UNARY_FUNCTIONS::STR_LEN:
-                *(T *)out = buf.num_codepoints();
-                break;
-        }
-    }
-};
-
-
-template <IMPLEMENTED_UNARY_FUNCTIONS f, const char* function_name>
+typedef bool (Buffer<ENCODING::UTF8>::*utf8_buffer_method)();
```

Comment:
nice, this is much simpler with the typedef, i'm glad the functor wasn't necessary in the end

Reply:
This one took my by far the longest - mostly to realize that this method pointers are 16 bytes long. But, yes, once I had it, I was pleased with it!